### ISSUE TITLE
Installation issue failed tests on Mac OS 

### ISSUE THREAD

**Issue by jharaldson** [vega/voyager#239]
Having major issues installing voyager, both on Mac OS as well as on Ubuntu Trusty.
This issue is raised with regard to the Mac OS installation, where three tests are failing.

Following the instructions:
https://github.com/vega/voyager

git clone https://github.com/vega/voyager
brew install node
cd voyager
npm install
npm start

```
$ npm start

> voyager@0.9.0 start /Users/a/sandbox/tmp/voyager
> gulp serve

[16:42:29] Using gulpfile ~/sandbox/tmp/voyager/gulpfile.js
[16:42:29] Starting 'styles'...
[16:42:29] Starting 'jshint'...
[16:42:29] gulp-inject 7 files into index.scss.
[16:42:30] Finished 'styles' after 552 ms
[16:42:30] Starting 'inject'...
[16:42:30] gulp-inject 1 files into index.html.
[16:42:30] gulp-inject 10 files into index.html.
[16:42:30] Finished 'jshint' after 575 ms
[16:42:30] Finished 'inject' after 210 ms
[16:42:30] Starting 'watch'...
[16:42:30] Finished 'watch' after 29 ms
[16:42:30] Starting 'serve'...
[16:42:30] Starting 'partials'...
[16:42:30] Finished 'serve' after 60 ms
[16:42:30] Finished 'partials' after 91 ms
[16:42:30] Starting 'test:auto'...
WARN `start` method is deprecated since 0.13. It will be removed in 0.14. Please use 
  server = new Server(config, [done])
  server.start()
instead.
[16:42:30] 'test:auto' errored after 46 ms
[16:42:30] Error: Cannot find module 'lolex'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.require.resolve (internal/module.js:23:19)
    at framework (/Users/a/sandbox/tmp/voyager/node_modules/karma-sinon-chai/index.js:26:40)
    at Array.invoke (/Users/a/sandbox/tmp/voyager/node_modules/di/lib/injector.js:75:15)
    at get (/Users/a/sandbox/tmp/voyager/node_modules/di/lib/injector.js:48:43)
    at /Users/a/sandbox/tmp/voyager/node_modules/karma/lib/server.js:138:20
    at Array.forEach (native)
    at Server._start (/Users/a/sandbox/tmp/voyager/node_modules/karma/lib/server.js:137:21)
    at invoke (/Users/a/sandbox/tmp/voyager/node_modules/di/lib/injector.js:75:15)
    at Server.start (/Users/a/sandbox/tmp/voyager/node_modules/karma/lib/server.js:102:18)
    at Object.oldServer.start (/Users/a/sandbox/tmp/voyager/node_modules/karma/lib/index.js:16:12)
    at runTests (/Users/a/sandbox/tmp/voyager/gulp/unit-tests.js:7:9)
    at Gulp.<anonymous> (/Users/a/sandbox/tmp/voyager/gulp/unit-tests.js:18:3)
    at module.exports (/Users/a/sandbox/tmp/voyager/node_modules/orchestrator/lib/runTask.js:34:7)
    at Gulp.Orchestrator._runTask (/Users/a/sandbox/tmp/voyager/node_modules/orchestrator/index.js:273:3)
    at Gulp.Orchestrator._runStep (/Users/a/sandbox/tmp/voyager/node_modules/orchestrator/index.js:214:10)
[BS] Access URLs:
 -----------------------------------------
       Local: http://localhost:3008/
    External: http://147.214.210.217:3008/
 -----------------------------------------
          UI: http://localhost:3009
 UI External: http://147.214.210.217:3009
 -----------------------------------------
[BS] Serving files from: .tmp/serve
[BS] Serving files from: src
[BS] Watching files...
```

Adding the following modules:
npm install lolex
npm install sinon
npm install sinon-chai

```
npm start

> voyager@0.9.0 start /Users/ejohara/sandbox/tmp/voyager
> gulp serve

[16:44:13] Using gulpfile ~/sandbox/tmp/voyager/gulpfile.js
[16:44:13] Starting 'styles'...
[16:44:13] Starting 'jshint'...
[16:44:13] gulp-inject 7 files into index.scss.
[16:44:13] Finished 'styles' after 559 ms
[16:44:13] Starting 'inject'...
[16:44:13] gulp-inject 1 files into index.html.
[16:44:13] gulp-inject 10 files into index.html.
[16:44:13] Finished 'inject' after 208 ms
[16:44:13] Starting 'watch'...
[16:44:13] Finished 'watch' after 26 ms
[16:44:13] Finished 'jshint' after 621 ms
[16:44:13] Starting 'serve'...
[16:44:14] Starting 'partials'...
[16:44:14] Finished 'serve' after 62 ms
[16:44:14] Finished 'partials' after 88 ms
[16:44:14] Starting 'test:auto'...
WARN `start` method is deprecated since 0.13. It will be removed in 0.14. Please use 
  server = new Server(config, [done])
  server.start()
instead.
03 05 2016 16:44:14.147:WARN [watcher]: Pattern "/Users/ejohara/sandbox/tmp/voyager/node_modules/sinon/pkg/sinon-timers.js" does not match any file.
03 05 2016 16:44:14.160:WARN [watcher]: Pattern "/Users/ejohara/sandbox/tmp/voyager/src/vendor/*.js" does not match any file.
03 05 2016 16:44:14.204:WARN [karma]: No captured browser, open http://localhost:9875/
03 05 2016 16:44:14.213:INFO [karma]: Karma v0.13.22 server started at http://localhost:9875/
03 05 2016 16:44:14.217:INFO [launcher]: Starting browser PhantomJS
[BS] Access URLs:
 -----------------------------------------
       Local: http://localhost:3008/
    External: http://147.214.210.217:3008/
 -----------------------------------------
          UI: http://localhost:3009
 UI External: http://147.214.210.217:3009
 -----------------------------------------
[BS] Serving files from: .tmp/serve
[BS] Serving files from: src
[BS] Watching files...
03 05 2016 16:44:15.307:INFO [PhantomJS 1.9.8 (Mac OS X 0.0.0)]: Connected on socket /#pQO1U3Qo8fJI7_EqAAAB with id 9675927

START:
  Service: Fields
    ✔ should test something
  Service: Visrec
    ✔ should do something
  Directive: encodingVariations
    ✔ should make hidden element visible
  Directive: fieldList
    ✖ should make hidden element visible
  Directive: fieldListItem
    ✖ should make hidden element visible
  Directive: functionList
    ✔ should have correct number of radio
LOG: '[Logging] ', 'FIELDS_CHANGE', '', Object{selected: [], list: []}
  Directive: visList
    ✖ should add vis list element

Finished in 0.209 secs / 0.118 secs

SUMMARY:
✔ 4 tests completed
✖ 3 tests failed

FAILED TESTS:
  Directive: fieldList
    ✖ should make hidden element visible
      PhantomJS 1.9.8 (Mac OS X 0.0.0)
    TypeError: 'undefined' is not an object (evaluating 'scope.field.name')
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/vega-lite-ui/vlui.js:1818
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:9073
        at invokeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:9079)
        at nodeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:8571)
        at delayedNodeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:8828)
        at compositeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:7965)
        at nodeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:8561)
        at delayedNodeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:8828)
        at compositeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:7965)
        at compositeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:7968)
        at publicLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:7845)
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:7986
        at controllersBoundTransclude (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:8593)
        at ngRepeatAction (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:28094)
        at $watchCollectionAction (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:16066)
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:16203
        at /Users/ejohara/sandbox/tmp/voyager/src/components/fieldlist/fieldlist.test.js:48

  Directive: fieldListItem
    ✖ should make hidden element visible
      PhantomJS 1.9.8 (Mac OS X 0.0.0)
    TypeError: 'undefined' is not an object (evaluating 'scope.field.name')
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/vega-lite-ui/vlui.js:1818
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:9073
        at invokeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:9079)
        at nodeLinkFn (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:8571)
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:8811
        at processQueue (/Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:15097)
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:15113
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:16359
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:16175
        at /Users/ejohara/sandbox/tmp/voyager/src/components/fieldlistitem/fieldlistitem.test.js:35

  Directive: visList
    ✖ should add vis list element
      PhantomJS 1.9.8 (Mac OS X 0.0.0)
    TypeError: 'undefined' is not a function (evaluating 'vl.shorthand.shortenFieldDefs(selectedFields)')
        at /Users/ejohara/sandbox/tmp/voyager/src/app/visrec/visrec.service.js:124
        at updateFields (/Users/ejohara/sandbox/tmp/voyager/src/components/vislist/vislist.js:55)
        at /Users/ejohara/sandbox/tmp/voyager/src/components/vislist/vislist.js:66
        at /Users/ejohara/sandbox/tmp/voyager/bower_components/angular/angular.js:16203
        at /Users/ejohara/sandbox/tmp/voyager/src/components/vislist/vislist.test.js:24
```

The end result is that web browser window is basically empty, as the screenshot shows.
Would this be due to incorrect package dependencies?

<img width="1846" alt="screenshot" src="https://cloud.githubusercontent.com/assets/7150893/14987282/9e8e50c6-114f-11e6-98e7-d9d8acefcf88.png">

**Comment by domoritz** [comment]
We are aware that the build does not work right now and will fix it soon (https://github.com/vega/polestar/issues/364). Please use the compiled version on gh-pages in the meantime.

**Comment by jharaldson** [comment]
Do you mean this one?
http://vega.github.io/voyager/

**Comment by domoritz** [comment]
Yeah. That page shows the content of the gh-pages branch (https://github.com/vega/voyager/tree/gh-pages). You can just checkout that branch or download the code directly. There is no need to compile anything. 

The main issue is that we didn't fix the versions of our dependencies to specific versions. We will fix that in coming releases very soon.

**Comment by keithhelfrich** [comment]
I've reached the same error, and I'm also finding that when I upload my own CSV to the compiled  version on gh-pages, it shows my data as the "chosen" data set.  But all charts still refer to the cars data set.

Is it possible to upload our own CSV to the compiled  version on gh-pages ?

Thanks!

**Comment by domoritz** [comment]
@keithhelfrich I believe that the issue is that the CSV file is not according to the expected format. Please make sure that there are no spaces in fields names, that you have a header and proper escaping. You might see an error in browser console.

**Comment by theharshbhatia** [comment]
@domoritz Hi, 
I am a software developer from India, interested in development of polestar. I tried cloning and installing polestar repo but it didn't work, @domoritz informed in the comments of issue(https://github.com/vega/voyager/issues/239) that the build is broken.
Please suggest a way for me to clone this repo to enable me to contribute in polestar development. I may start with fixing some small issue or help in creating small new feature in polestar / vega.

**Comment by domoritz** [comment]
Hi @theharshbhatia That's fantastic. I think at this point I have to ask you to wait for a few weeks. @kanitw and I need to spend some time to get polestar and voyager up to vega-lite 1.0. Once that is done the build will work again and we would love to get your help.

**Comment by bobbyangelov** [comment]
+1 on this issue, going to be using `gh-pages` at the moment then. Still, thank you for the great tool, looking forward towards the future versions, maybe I could help out with something as well!

**Comment by rsiebeling** [comment]
+1 for this issue

**Comment by kanitw** [comment]
This should be fixed as of 3fcf1850efde6e9ccfc3806513e8c50e2be3a459.  
If you still run into a problem, please create another issue. 

<img width="613" alt="1__vy__node_" src="https://cloud.githubusercontent.com/assets/111269/17339380/76d07dbe-58a0-11e6-84b6-6c3da2039a8f.png">

**Comment by domoritz** [comment]
❇️
